site/public/js/forum.js

```diff
@@ -1335,32 +1334,36 @@ function toggleLike(post_id, current_user) {
             const likeIconSrc = document.getElementById(`likeIcon_${post_id}`);
             let likeIconSrcElement = likeIconSrc.src;
 
-            const theme = localStorage.getItem('theme');
             if (liked==='unlike') {
-                if (theme==='light' && likeIconSrcElement.endsWith('/img/on-duck-button.svg')) {
-                    likeIconSrcElement = likeIconSrcElement.replace('on-duck-button.svg', 'light-mode-off-duck.svg');
-                }
-                else {
-                    likeIconSrcElement = likeIconSrcElement.replace('on-duck-button.svg', 'light-mode-off-duck.svg');
+                likeIconSrcElement = likeIconSrcElement.replace('on-duck-button.svg', 'light-mode-off-duck.svg');
+                if (userGroup === 1 || userGroup === 2 || userGroup === 3) {
+                    if (taLiked>0) {
+                        $(`#likedByInstructor_${post_id}`).show();
+                    }
+                    else {
+                        $(`#likedByInstructor_${post_id}`).hide();
+                    }
                 }
+
```

Comment:
Don't we only care if taLiked value is true? Why does it matter what role/group we are here? I think you should be able to get rid of the first if statement.